ui/app/contexts/metametrics.new.js

```diff
@@ -30,12 +32,39 @@ export const MetaMetricsContext = createContext(() => {
 
 const PATHS_TO_CHECK = Object.keys(PATH_NAME_MAP)
 
+function useSegmentContext () {
+  const match = useRouteMatch({ path: PATHS_TO_CHECK, exact: true, strict: true })
+  const locale = useSelector(getCurrentLocale)
+  const txData = useSelector(txDataSelector) || {}
+  const confirmTransactionOrigin = txData.origin
+
+  const referrer = confirmTransactionOrigin ? {
+    url: confirmTransactionOrigin,
+  } : undefined
+
+  return {
+    app: {
+      version: `${global.platform.getVersion()}-${process.env.METAMASK_ENVIRONMENT}`,
+      name: 'MetaMask Extension',
+    },
+    locale,
+    page: {
+      path: match.path,
+      title: PATH_NAME_MAP[match],
+      url: match.path,
+    },
+    referrer,
```

Comment:
Hey this is a really interesting idea - using the dapp that triggered the confirmation as the referrer. Not _exactly_ what this field is usually used for, but it feels right semantically :thinking:. I think I like this.